### PR TITLE
fix(web): maintenance of focus when changing keyboard via Toolbar UI

### PR DESCRIPTION
**Pull request by jahorton**
Fixes https://github.com/keymanapp/keyman/issues/9404.

It turns out that the issue was due to a small mishandling of state management when in a focus-maintenance state.

## User Testing

TEST_TOOLBAR_KEYBOARD_SWAP:  Using the "Desktop UI module testing" test-page link on a desktop form-factor device...
1. Set the page to use the Toolbar UI.
2. Ensure that any KMW keyboard is active and type a couple of keys.
3. Use the toolbar UI to swap to the Khmer language, then select Khmer Angkor from the keyboard-selection dropdown.
4. With no further mouse interaction anywhere on the page, type the `k` key, then the `a` key.  Verify that Khmer characters are output.

### CODE REVIEW

**Comment by bharanidharanj** [comment]
# Test Results

* **TEST_TOOLBAR_KEYBOARD_SWAP (PASSED):** Tested this PR as per the instructions and here is my observation: 1. After typing k, then a key from the keyboard, it outputs khmer text 'កា'. Seems to be working fine. Thanks.

File: web/src/app/browser/src/contextManager.ts [hunk 1]
```diff
@@ -211,6 +211,12 @@ export default class ContextManager extends ContextManagerBase<BrowserConfigurat
     const originalTarget = this.activeTarget; // may differ, depending on focus state.
 
     if(target == originalTarget) {
+      // A focus state may have .currentTarget as null at this stage; if the func
+      // is being called with a non-null parameter, we want this SET.
```
Comment (mcdurdin) [inline web/src/app/browser/src/contextManager.ts:215]:
```suggestion
      // A focus state may have .currentTarget as null at this stage; if the func
      // is being called with a non-null parameter, we want this SET. #9404
```

**Review by mcdurdin — APPROVED** [review]
LGTM. I moved the issue into the Keyman repo since it wasn't a keymanweb.com bug.

File: web/src/app/browser/src/contextManager.ts [hunk 2]
```diff
@@ -211,6 +211,12 @@ export default class ContextManager extends ContextManagerBase<BrowserConfigurat
...
+      if(target) {
```
Comment (mcdurdin) [inline web/src/app/browser/src/contextManager.ts:216]:
```suggestion
      if(originalTarget) {
```

It comes to the same thing but is less cognitive load to read I think!

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.153-alpha](https://keyman.com/downloads/releases/alpha/17.0.153)

**Pull request merged**